bluemira/base/parameter.py

```diff
@@ -239,6 +274,36 @@ def __init__(
             self._value_history = []
             self._source_history = []
 
+    def __dir__(self):
+        """
+        Add missing methods
+        """
+        return sorted(
```

Comment:
Aren't `set`s already sorted? Would a straight conversion to `list` work?

Reply:
yes you're right, I will just use list